src/gui/qgsmaptooldigitizefeature.cpp

```diff
@@ -341,9 +341,9 @@ void QgsMapToolDigitizeFeature::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
           QgsGeometry featGeom = f->geometry();
           int avoidIntersectionsReturn = featGeom.avoidIntersections( avoidIntersectionsLayers );
           f->setGeometry( featGeom );
-          if ( avoidIntersectionsReturn == 1 )
+          if ( avoidIntersectionsReturn == 3 )
           {
-            //not a polygon type. Impossible to get there
+            emit messageEmitted( tr( "The feature has been added, but at least one geometry intersected is invalid. You should fix geometries." ), Qgis::Warning );
```

Comment:
```suggestion
            emit messageEmitted( tr( "The feature has been added, but at least one geometry intersected is invalid. These geometries must be manually repaired." ), Qgis::Warning );
```

Reply:
Is the 'manually' mandatory? I think simply stating that they should be repair is enough.

Reply:
I think so -- I want to start discouraging users from thinking that the auto repair tools are a magic wand and the correct solution to use. I'd prefer our message to be "you should fix them yourselves, in a way which is appropriate for the work you are conducting and using your knowledge of the features themselves". Hence "manually"!

Reply:
Ok then, it's just that sometime when I add new geometries on a dataset without issue I get the processing error. And I'm always puzzled and annoyed what adding a few simply polygons without overlap seemingly invalidate things and the current fixers seem to do the work, whitout any geometry change. (Hence why the other PR). And also somethings processing can create invalid geometries. Dealing with such can be annoying and time consuming...